### PR TITLE
Refactoring done in various parts of code.

### PR DESCRIPTION
**Pull request by jaymulani**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MJAVADOC) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MJAVADOC-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MJAVADOC-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify -Prun-its` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by jaymulani** [comment]
.

File: pom.xml [hunk 1]
```diff
@@ -380,6 +380,16 @@ under the License.
       <version>${slf4jVersion}</version>
       <scope>test</scope>
     </dependency>
+      <dependency>
+          <groupId>org.apache.maven.plugins</groupId>
```
Comment (elharo) [inline pom.xml:384]: indentation is off

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractFixJavadocMojo.java [hunk 2]
```diff
@@ -873,19 +876,11 @@ private ClassLoader getProjectClassLoader() throws MojoExecutionException {
...
-            return;
-        }
-
-        File javaFile;
-        try {
-            javaFile = Paths.get(javaClass.getSource().getURL().toURI()).toFile();
-        } catch (URISyntaxException e) {
-            throw new MojoExecutionException(e.getMessage());
-        }
+    // Refactoring : Extract Method
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractFixJavadocMojo.java:881]: needless comment

File: src/main/java/org/apache/maven/plugins/javadoc/AbstractFixJavadocMojo.java [hunk 3]
```diff
@@ -1205,11 +1227,23 @@ private boolean fixFieldComment(
      * <font color="#2a00ff">&#34;value&#34;</font><font color="#000000">;</font>
      * </code>
      *
-     * @param stringWriter not null
-     * @param field        not null
-     * @param indent       not null
+//     * @param stringWriter not null
```
Comment (elharo) [inline src/main/java/org/apache/maven/plugins/javadoc/AbstractFixJavadocMojo.java:1230]: why?

**Review by elharo — CHANGES_REQUESTED** [review]
There are a lot of small issues here, but overall I don't see the reason for any of this.

**Pull request closed without merging**